src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -69,14 +77,22 @@ export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<StoredProce
         return this.parent.getContainerClient(client);
     }
 
-    private validateName(name: string): string | null | undefined {
+    private async validateStoredProcedureName(name: string, currStoredProcedureNames: String[]): Promise<string | null | undefined> {
         if (name) {
-            if (name.indexOf("/") !== -1 || name.indexOf("\\") !== -1 || name.indexOf("?") !== -1 || name.indexOf("#") !== -1) {
-                return "Id contains illegal chars: /,\\,?,#";
+            if (name.length < 1 || name.length > 255) {
+                return "Name has to be between 1 and 255 chars long";
+            }
+            if (/[/\\?#&]/.test(name)) {
+                return localize("illegalChars", "Name contains illegal chars: /,\\,?,#,&");
```

Comment:
nit: I think adding a space between each invalid character makes this look a bit more approachable

| Current | Suggested |
|-|-|
| <img width="523" alt="Screen Shot 2021-02-09 at 5 02 15 PM" src="https://user-images.githubusercontent.com/22795803/107449577-a81f7680-6af8-11eb-9cec-3d3ee8d05b61.png"> | <img width="523" alt="Screen Shot 2021-02-09 at 5 00 29 PM" src="https://user-images.githubusercontent.com/22795803/107449586-ac4b9400-6af8-11eb-974b-55a455e5ea58.png"> |

Reply:
Should we keep the commas?